File: tests/fixtures/anti-patterns/typescript/long-method-vulnerable.ts
/**
 * Test Fixture: Long Method Vulnerable Pattern (TypeScript)
 *
 * This file contains a function with many statements.
 * Expected detections: >=1 violation for AP-005
 * Rule ID: AP-005
 * Severity: MEDIUM (info)
 */

interface UserData {
  userId?: number;
  username?: string;
  email?: string;
  password?: string;
  firstName?: string;
  lastName?: string;
}

// VULNERABLE: Function with 15+ statements
export function veryLongProcessingFunction(data: UserData): Record<string, any> {
  const result: Record<string, any> = {};
  const errors: string[] = [];
  const warnings: string[] = [];

  if (!data) {
    errors.push("Data is empty");
    return { status: "error", errors };
  }

  const userId = data.userId;
  const username = data.username;
  const email = data.email;
  const password = data.password;
  const firstName = data.firstName;
  const lastName = data.lastName;

  if (!userId) {
    errors.push("Missing userId");
  }

  if (!username) {
    errors.push("Missing username");
  }

  if (!email) {
    errors.push("Missing email");
  }

  if (!password) {
    errors.push("Missing password");
  }

  result.userId = userId;
  result.username = username?.toLowerCase();
  result.email = email?.toLowerCase();
  result.firstName = firstName;
  result.lastName = lastName;
  result.status = "success";
  result.warnings = warnings;

  return result;
}

File: tests/fixtures/anti-patterns/csharp/LongMethodVulnerable.cs
/**
 * Test Fixture: Long Method Vulnerable Pattern (C#)
 *
 * This file contains a method with many statements.
 * Expected detections: >=1 violation for AP-005
 * Rule ID: AP-005
 * Severity: MEDIUM (info)
 */

using System;
using System.Collections.Generic;

namespace TestFixtures.AntiPatterns
{
    public class LongMethodVulnerable
    {
        // VULNERABLE: Method with 15+ statements
        public Dictionary<string, object> VeryLongProcessingMethod(Dictionary<string, object> data)
        {
            var result = new Dictionary<string, object>();
            var errors = new List<string>();
            var warnings = new List<string>();

            if (data == null)
            {
                errors.Add("Data is empty");
                result["status"] = "error";
                result["errors"] = errors;
                return result;
            }

            var userId = data.GetValueOrDefault("userId");
            var username = data.GetValueOrDefault("username");
            var email = data.GetValueOrDefault("email");
            var password = data.GetValueOrDefault("password");
            var firstName = data.GetValueOrDefault("firstName");
            var lastName = data.GetValueOrDefault("lastName");

            if (userId == null)
            {
                errors.Add("Missing userId");
            }

            if (username == null)
            {
                errors.Add("Missing username");
            }

            if (email == null)
            {
                errors.Add("Missing email");
            }

            if (password == null)
            {
                errors.Add("Missing password");
            }

            result["userId"] = userId;
            result["username"] = username?.ToString()?.ToLower();
            result["email"] = email?.ToString()?.ToLower();
            result["firstName"] = firstName;
            result["lastName"] = lastName;
            result["status"] = "success";
            result["warnings"] = warnings;

            return result;
        }
    }
}
